File: src/SampleCRM.Web/Services/OrderService.cs
using OpenRiaServices.DomainServices.Hosting;
using OpenRiaServices.DomainServices.Server;
using SampleCRM.Web.Attributes;
using SampleCRM.Web.Models;
using System;
using System.Data.Entity.Migrations;
using System.Linq;

namespace SampleCRM.Web
{
    [EnableClientAccess]
    public class OrderService : SampleCRMService
    {
        [Query]
        public IQueryable<Orders> GetOrders(string search) =>
            _context.Orders.OrderByDescending(o => o.OrderDateUTC)
                           .Where(x => x.OrderID.ToString().Contains(search) 
                                       || search == "")
                           .AsQueryable();

        [Query]
        public IQueryable<Orders> GetLatestOrders(int limit, string search) => 
            GetOrders(search).Take(limit);

        [Query]
        public IQueryable<Orders> GetOrdersOfCustomer(long customerId, string search) => 
            GetOrders(search).Where(x => x.CustomerID == customerId);

        [Delete]
        [RestrictAccessReadonlyMode]
        public void DeleteOrder(Orders order)
        {
            var dorder = _context.Orders.FirstOrDefault(x => x.OrderID == order.OrderID);
            if (dorder == null)
                return;

            _context.Orders.Remove(dorder);
            _context.SaveChanges();
        }

        [Insert]
        [RestrictAccessReadonlyMode]
        public void InsertOrder(Orders order)
        {
            order.OrderID = new Random().Next((int)Math.Pow(10, 12), (int)Math.Pow(10, 13) - 1);
            if (order.OrderID < 0)
                order.OrderID *= -1;

            order.OrderDateUTC = order.LastModifiedOnUTC = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
            _context.Orders.Add(order);
#if DEBUG
            var validationResult = _context.Entry(order).GetValidationResult();
            if (validationResult.ValidationErrors.Any())
            {
                Console.WriteLine($"Validation Error in InsertOrder: {validationResult.ValidationErrors.FirstOrDefault().PropertyName} {validationResult.ValidationErrors.FirstOrDefault().ErrorMessage}");
            }
#endif
            _context.SaveChanges();
        }

        [Update]
        [RestrictAccessReadonlyMode]
        public void UpdateOrder(Orders order)
        {
            order.LastModifiedOnUTC = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
#if DEBUG
            var validationResult = _context.Entry(order).GetValidationResult();
            if (validationResult.ValidationErrors.Any())
            {
                Console.WriteLine($"Validation Error in InsertOrder: {validationResult.ValidationErrors.FirstOrDefault().PropertyName} {validationResult.ValidationErrors.FirstOrDefault().ErrorMessage}");
            }
#endif
            _context.Orders.AddOrUpdate(order);
            _context.SaveChanges();
        }
    }
}

File: src/SampleCRM.Web/Services/OrderService.vb
Imports OpenRiaServices.DomainServices.Hosting
Imports OpenRiaServices.DomainServices.Server
Imports SampleCRM.Web.Attributes
Imports System
Imports System.Data.Entity
Imports System.Data.Entity.Migrations
Imports System.Linq

Namespace SampleCRM.Web
    <EnableClientAccess>
    Public Class OrderService
        Inherits SampleCRMService

        <Query>
        Public Function GetOrders() As IQueryable(Of Orders)
            Return _context.Orders
        End Function

        <Query>
        Public Function GetLatestOrders(ByVal limit As Integer) As IQueryable(Of Orders)
            Return _context.Orders.OrderByDescending(Function(x) x.OrderDate).Take(limit)
        End Function

        <Query>
        Public Function GetOrdersOfCustomer(ByVal customerId As Long) As IQueryable(Of Orders)
            Return _context.Orders.Where(Function(x) x.CustomerID = customerId)
        End Function

        <Delete>
        <RestrictAccessDeveloperMode>
        Public Sub DeleteOrder(ByVal order As Orders)
            Dim dorder = _context.Orders.FirstOrDefault(Function(x) x.OrderID = order.OrderID)
            If dorder Is Nothing Then Return
            _context.Orders.Remove(dorder)
            _context.SaveChanges()
        End Sub

        <Insert>
        <RestrictAccessDeveloperMode>
        Public Sub InsertOrder(ByVal order As Orders)
            order.OrderID = New Random().[Next](CInt(Math.Pow(10, 12)), CInt(Math.Pow(10, 13)) - 1)
            If order.OrderID < 0 Then order.OrderID = order.OrderID * -1
            order.LastModifiedOn = Date.Now.ToString()
            order.OrderDate = order.LastModifiedOn
            _context.Orders.Add(order)
            _context.SaveChanges()
        End Sub

        <Update>
        <RestrictAccessDeveloperMode>
        Public Sub UpdateOrder(ByVal order As Orders)
            order.LastModifiedOn = Date.Now.ToString()
            _context.Orders.AddOrUpdate(order)
            _context.SaveChanges()
        End Sub
    End Class
End Namespace
